medmodels/matching/algorithms/classic_distance_models.py

```diff
@@ -90,10 +87,6 @@ def nearest_neighbor(
             )
             matched_control.append(new_row)
 
-            # For the k:1 matching don't consider the chosen row any more.
-            control_array_full = np.delete(control_array_full, neighbor_index, 0)
-            control_array = np.delete(control_array, neighbor_index, 0)
-
     return pl.concat(matched_control, how="vertical")
```

Comment:
why not

```python
return pl.DataFrame(
    [
        control_array_full[neighbor_index]
        for neighbor_indices in indices
        for neighbor_index in neighbor_indices
    ],
    orient="row",
    schema=treated_set.columns,
)
```

instead of the for loops and concatination

Reply:
You are right, better idea!